spec.md

```diff
@@ -15,4 +16,574 @@ The key words "unspecified", "undefined", and "implementation-defined" are to be
+| RPC/gRPC               | [Remote Procedure Call](https://en.wikipedia.org/wiki/Remote_procedure_call).                                         |
+| Node              | A host where the user workload will be running, uniquely identifiable from the perspective of a Plugin by a node ID. |
+| COSI Driver(aka Plugin)           | Aka “plugin implementation”, a gRPC endpoint that implements the COSI Services.                                        |
+| Sidecar (aka Plugin Sidecar) | A COSI process that interfaces with the Plugin.                                                       |
+|COSI Node Adapter (aka CSI Driver) | A process that accepts gRPC calls from the Kubelet notifying it when the workload has been started and when it is terminating. |
+| Provisioner | A generic term meant to describe the combination of a sidecar and COSI driver. "Provisioning" a bucket can mean creating a new bucket or granting access to an existing bucket.                                                       |
+| Workload          | The atomic unit of "work" scheduled by a COSI System. This MAY be a container or a collection of containers.                   |
+
+## Objective
+
+To define a standard “Container Object Storage Interface” (COSI) that will enable storage vendors (SP) to develop a plugin once and have it work across multiple Kubernetes Clusters and on other COs.
+
+### Goals in MVP
+
+The Container Object Storage Interface (COSI) will:
+
+* Enable SP authors to write one COSI compliant Plugin that “just works” across all COs that implement COSI.
+* Define API (RPCs) that enable:
+  * Dynamic provisioning and de-provisioning of a bucket.
+  * Ability to define bucket access control mechanisms
+  * Consumption of both existing and new buckets created on SP.
+  * Define plugin protocol RECOMMENDATIONS.
+  * Describe a process by which a admin configures a Plugin.
+  * COSI deployments and templates. 
+
+### Non-Goals in MVP
+
+The Container Object Storage Interface (COSI) explicitly will not define, provide, or dictate:
+
+* Specific mechanisms by which a Plugin Sidecar manages the lifecycle of a Plugin, including:
+  * How to deploy, install, upgrade, uninstall, monitor, or respawn (in case of unexpected termination) Plugins.
+* A first class message structure/field to represent "grades of object storage" (aka "bucket class").
+* Protocol-level authentication and authorization.
+* Packaging of a Plugin.
+
+## Solution Overview
+
+This specification defines an interface along with the minimum operational and packaging recommendations for a storage provider (SP) to implement a COSI compatible plugin.
+The interface declares the RPCs that a plugin MUST expose: this is the **primary focus** of the COSI specification.
+Any operational and packaging recommendations offer additional guidance to promote cross-COSI system compatibility.
+
+### Architecture
+
+The primary focus of this specification is on the **protocol** between a COSI System and a Plugin.
```

Comment:
consider changing to:
`The primary focus of this specification is the RPC **protocol** between a COSI System and a Plugin.`

Reply:
fixed.